docs/extend/extend-apiml/api-mediation-oidc-authentication.md

```diff
@@ -0,0 +1,95 @@
+# Zowe API Mediation Layer OIDC Authentication
+
+The OpenID/Connect ([OIDC](https://openid.net/specs/openid-connect-core-1_0.html)) protocol adds an authentication layer on top of the [OAuth2](https://www.rfc-editor.org/rfc/rfc6749) Authorization protocol.
+OIDC authentication, together with the z/OS [Identity Propagation](https://www.redbooks.ibm.com/redbooks/pdfs/sg247850.pdf) mechanism, is the foundation of the API ML Identity Federation.
+Zowe API ML can be configured to authenticate users by accepting Access Tokens issued by an external OIDC provider.
+This configuration is useful in advanced deployments of Zowe where client applications need to access mainframe as well as enterprise/distributed systems while simultaneously offering single sign-on (SSO) across system boundaries.  
+
+Review this topic for details about the API ML OIDC authentication functionality and how to configure this feature. 
+
+- [Usage](#usage)
+- [Authentication flow](#authentication-flow)
+- [Prerequisites](#prerequisites)
+  * [OIDC provider](#oidc-provider)
+  * [ESM configuration](#esm-configuration)
+- [API ML configuration](#api-ml-configuration)
+- [Troubleshooting](#troubleshooting)
+
+## Usage
+The OIDC protocol is used by API ML client applications to verify the identity of a user with a distributed OIDC provider trusted by the mainframe security manager.
+Upon successful user login, the OIDC provider grants the client application an Access Token along with an Identity Token.
+The client application can pass this OIDC token with subsequent requests to mainframe services routed through the API ML Gateway.
+The API ML Gateway then validates the OIDC Access Token. If the token is valid, the user identity from that token is mapped to the mainframe identity of the user.
+This API ML Gateway can then route the request to target API services with mainframe user credentials (JWT or a Passticket) according to the service's authentication schema configuration.
+
+## Authentication Flow
+The following diagram illustrates the interactions between the participants of the OIDC based API ML authentication process.
+
+<img src={require("../../images/api-mediation/apiml-oidc-auth-seq.png").default} alt="APIML OIDC Workflow" width="700"/>
+
+* When a user wants to access mainframe resources or services using the client application without a valid authentication / access token, the client redirects the user agent to the login end-point of the distributed OIDC provider. 
+* The user is asked to provide valid credentials (authentication factors).
+* After successful validation of all authentication factors, the OIDC provider grants the client an access token.
+* The client can then request from API ML Gateway the needed mainframe resources presenting the access token in the request. 
+* The Gateway validates the access token at the provider's OIDC/introspection end-point. If the access token is validated, the outcome is cached for a short time. 
```

Comment:
How long is a short time?

Reply:
Provided the default value (20 sec) as configured in hacache.xml